tests/integration/cors.js

```diff
@@ -6,9 +6,16 @@ describe('Simple CORS request', function () {
     it('should include the correct headers', function (done) {
         server
         .get('/https://google.com')
+        .set('Origin', 'http://example.com')
```

Comment:
Requiring origin header should be in a separate PR as mentioned above (also, would error 412 be better here?)